tools/config/__init__.py

```diff
@@ -600,15 +600,26 @@ def _get_cmsis_part(self):
                                   "arm_pack_manager index.")
         return cache.index[self.target.device_name]
 
-    def _get_mem_specs(self, memories, cmsis_part, exception_text):
-        for memory in memories:
-            try:
-                size = cmsis_part['memory'][memory]['size']
-                start = cmsis_part['memory'][memory]['start']
-                return (start, size)
-            except KeyError:
-                continue
-        raise ConfigException(exception_text)
+    @staticmethod
+    def _memory_ordering(memory):
+        return (memory['default'], memory['size'], memory['start'])
+
+    def _get_mem_specs(self, permissions, cmsis_part):
+        all_matching_memories = [
+            memory for memory in cmsis_part['memories'].values()
+            if all(memory['access'][perm] for perm in permissions)
```

Comment:
Is it guarenteed that the key `perm` will be in the dictionary `memory['access']`? If not, this would be a safer alternative (no risk of `KeyError`):
```suggestion
            if all(memory['access'].get(perm) for perm in permissions)
```

Reply:
Technically it's guaranteed. OTOH, it's safer to do it the way you recommend.